mqtt/src/test/scala/akka/stream/alpakka/mqtt/scaladsl/MqttSourceSpec.scala

```diff
@@ -169,5 +170,42 @@ class MqttSourceSpec
         elem2.futureValue shouldBe MqttMessage(topic1, ByteString("ohi"))
       }
     }
+
+    "support will message" in {
+      import system.dispatcher
+
+      val ks = Tcp()
+        .bind("localhost", 1337)
+        .runWith(Sink.head)
+        .map(
+          _.handleWith(Tcp().outgoingConnection("localhost", 1883).viaMat(KillSwitches.single)(Keep.right))
+        )
```

Comment:
I looked at the test failures and there is a race between starting a proxy and starting a testator connection. You should keep the server binding materialized value from the proxy and run the testator source only when proxy is up and running. Something in the lines of this:

```scala
val (binding, connection) = Tcp().bind("localhost", 1337).toMat(Sink.head)(Keep.both).run()

val ks = connection.map(
  _.handleWith(Tcp().outgoingConnection("localhost", 1883).viaMat(KillSwitches.single)(Keep.right))
)

// wait for the proxy to start
whenReady(binding) { ... }
```

So it was not problem with message retaining after all. :)